### PR TITLE
RTParams

### PR DESCRIPTION
**Pull request by marcbonnici**
Dependant on PR: https://github.com/ARM-software/devlib/pull/157

### CODE REVIEW

File: wa/framework/target/runtime_config.py [hunk 1]
```diff
@@ -809,3 +809,94 @@ def _get_common_idle_values(self):
                 if state.name not in common_idle_states:
                     common_idle_states.append(state)
         return common_idle_states
+
+
+class ScreenOrientation(object):
+
+    def __init__(self):
+        self.values = {'NATURAL': 0, 'LEFT': 1, 'INVERTED': 2, 'RIGHT': 3}
```
Comment (setrofim) [inline wa/framework/target/runtime_config.py]: Use an `enum` here?

File: wa/framework/target/runtime_config.py [hunk 2]
```diff
@@ -809,3 +809,94 @@ def _get_common_idle_values(self):
...
+                              description="""
+                              Specify the screen orientation for the device
+                              """)
+
+    def check_target(self):
+        if self.target.os != 'android':
+            raise ConfigError('Target does not appear to be running Android')
+
+    def validate_parameters(self):
+        pass
```
Comment (setrofim) [inline wa/framework/target/runtime_config.py]: Should check that `self.config['brightness']` is between `0` and `255`, if set?

File: wa/framework/target/runtime_config.py [hunk 3]
```diff
@@ -809,3 +809,89 @@ def _get_common_idle_values(self):
                 if state.name not in common_idle_states:
                     common_idle_states.append(state)
         return common_idle_states
+
+
+class ScreenOrientation(object):
```
Comment (setrofim) [inline wa/framework/target/runtime_config.py]: `ScreenOrientation = enum(['NATURAL', 'LEFT', 'INVERTED', 'RIGHT'])` instead?

Comment (marcbonnici) [inline wa/framework/target/runtime_config.py]: Updated

File: wa/utils/serializer.py [hunk 4]
```diff
@@ -157,6 +165,9 @@ def _wa_regex_representer(dumper, data):
     text = '{}:{}'.format(data.flags, data.pattern)
     return dumper.represent_scalar(_regex_tag, text)
 
+def _wa_level_representer(dumper, data):
+    text = '{}:{}'.format(data.name, data.level)
+    return dumper.represent_scalar(_regex_tag, text)
```
Comment (setrofim) [inline wa/utils/serializer.py]: `_regex_tag` --> `_level_tag`

File: wa/utils/types.py [hunk 5]
```diff
@@ -554,6 +554,10 @@ class MyEnum(object):
 
     class Enum(object):
 
+        class __metaclass__(type):
```
Comment (setrofim) [inline wa/utils/types.py]: Why not just define `__str__` directly for `Enum`?

File: wa/utils/serializer.py (hunk 4 above)
Comment (marcbonnici) [inline wa/utils/serializer.py]: Fixed

**Pull request merged**